### ISSUE TITLE
Disponibilité de l'extension sur Chrome

### ISSUE THREAD

**Issue by RmbxCh** [lovasoa/ophirofox#236]
Bonjour,

Sur le webstore de Chrome, il y a cette alerte :

![image](https://github.com/user-attachments/assets/db77e3b4-958b-44a0-a577-c0d89ec33c21)

Est-ce que vous avez plus d'informations de la part de Chrome ? Est-ce lié à Manifest V3 ?

Bonne journée!

**Comment by lovasoa** [comment]
Oui, c'est lié à manifest v3. Le problème c'est que firefox n'est pas complètement compatible avec manifest v3. Si on migre, il faut maintenir deux extensions différentes pour Chrome et Firefox, ou écrire du code pour compiler deux versions différentes de l'extension.

**Comment by RmbxCh** [comment]
Ok, merci pour l'info!
Et vous savez si Chrome continuera à mettre l'extension à jour malgré tout ? (La dernière MAJ de l'extension n'est pas disponible pour le moment)

**Comment by lovasoa** [comment]
Non, google veut supprimer toutes les extensions manifest v2. Manifest v2 permet d'efficacement bloquer les publicités dans le navigateur, et google veut empêcher les bloqueurs de pubs de fonctionner correctement. 

https://arstechnica.com/gadgets/2024/05/google-starts-deprecating-older-more-capable-chrome-extensions-next-week/

https://www.eff.org/deeplinks/2021/12/googles-manifest-v3-still-hurts-privacy-security-innovation

https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline

**Comment by lnoss** [comment]
La dépréciation de MV2 dans Google Chrome sans retour débute. https://x.com/cpojer/status/1845774326324432935

Quels sont les axes d'améliorations acceptables pour la taille du projet ? Est-ce que l'extension reste à la _mano_ sur ses bases actuelles ? Peut-on envisager d'autres évolutions comme se rebaser par dessus un framework comme [Plasmo](https://docs.plasmo.com/) ?

**Comment by lovasoa** [comment]
Tout est acceptable à condition que ce soit maintenable dans le temps.
Je n'ai rien contre la migration si elle est facile, et qu'elle n'ajoute pas une pléthore de dépendances à mettre à jour régulièrement. 
Honnêtement, je pense que la migration à Manifest v3 directement est probablement plus facile.

**Comment by lnoss** [comment]
SI j'ai bien compris, on a deux gros mois devant nous pour y réfléchir et agir. Cela laisse du temps pour d'autres avis.

**Comment by lnoss** [comment]
Google est actuellement très poussif sur YouTube à l'approche du passage obligatoire à MV3 pour les internautes utilisant Google Chrome. Il y a un travail conséquent de leurs ingénieurs pour neutraliser les bloqueurs de pub sur YouTube ; ce qui devrait mieux facilité avec MV3.

Je n'ai pas avancé sur le sujet, mais la dépréciation totale semble donc imminente. Je vais me pencher dessus, mais si quelqu'un y travaille, qu'il se présente !

**Comment by Write** [comment]
@lnoss En effet, je pense qu'une version MV3 est vraiment nécessaire. La version sur le store semble ne plus être mise à jour depuis le 14 mai 2024. J'imagine que c'est parce que Chrome n'acceptait plus les mises à jour depuis cette date-là. Je ne sais pas si c'était déjà à cause du MV3 à l'époque cependant.

![Image](https://github.com/user-attachments/assets/4515ca4c-3e20-4ead-8e66-01c01d8862a8)

**Comment by D4llo** [comment]
Perso j'ai décidé de ne pas migrer mes extensions sur Chrome et d'uniquement supporter Firefox. Je pense qu'étant le monstrueur GAFAM que Google est, il est aussi de notre devoir de dev de ne pas alimenter son monopole sur le web.

Cela dit le code étant libre, rien n'empêche une personne motivée de forker et de s'occuper de maintenir une extension pour Chrome.

**Comment by Write** [comment]
L'extension a été mise à jour sur le Chrome store, bien que le bouton est indisponible (car mv2).

Ceci permet de l'installer tout de même sur les autres navigateurs supportants les extensions Chrome et autorisant encore le MV2.

Sur Chrome il est pour le moment encore possible de l'installer depuis le store en faisant clique droit, inspecter l'element, et en retirant la balise "disabled" du bouton d'ajout.

**Comment by mxgnr** [comment]
Hello 👋 

Cela fait un petit moment que je voulais proposer mon aide pour ce plugin très utile, et après quelques essais concluants, je peux vous annoncer que mon PoC sur le manifest V3 est fonctionnel sur Chrome et Firefox ! Vous pouvez le tester en local à partir de cette branche : https://github.com/mxgnr/ophirofox/tree/mv3

J'attire votre attention sur un sujet particulier qui a nécessité pas mal de modification : celui des domaines en referer. Je n'ai pas pu tester cette fonctionnalité vu que je ne possède pas de compte associé mais je serai ravi de le faire si quelqu'un serait en capacité de m'en fournir un ou plusieurs. 

Sinon, l'auth plus traditionnelle directement via URL semble fonctionner correctement, y compris la configuration d'options ainsi que le menu contextuel. 

N'hésitez pas si vous avez des retours ;)

**Comment by Write** [comment]
Absolument magnifique, merci @mxgnr
